src/sci/detci.h

```diff
@@ -0,0 +1,166 @@
+#include "integrals/integrals.h"
+#include "base_classes/active_space_method.h"
+#include "base_classes/rdms.h"
+#include "sparse_ci/sparse_ci_solver.h"
+#include "sparse_ci/ci_reference.h"
+#include "integrals/active_space_integrals.h"
+#include "sparse_ci/determinant.h"
+#include "sparse_ci/sigma_vector.h"
+
+namespace forte {
+class DETCI : public ActiveSpaceMethod {
+  public:
+    /**
+     * @brief DETCI Constructor
+     * @param state The state info (symmetry, multiplicity, na, nb, etc.)
+     * @param nroot Number of roots of interests
+     * @param scf_info SCF information
+     * @param options Forte options
+     * @param mo_space_info MOSpaceInfo
+     * @param as_ints Active space integrals
+     */
+    DETCI(StateInfo state, size_t nroot, std::shared_ptr<SCFInfo> scf_info,
+          std::shared_ptr<ForteOptions> options, std::shared_ptr<MOSpaceInfo> mo_space_info,
+          std::shared_ptr<ActiveSpaceIntegrals> as_ints);
+
+    /// Compute the energy
+    double compute_energy() override;
+
+    /// RDMs override
+    std::vector<RDMs> rdms(const std::vector<std::pair<size_t, size_t>>& root_list,
+                           int max_rdm_level) override;
+
+    /// Transition RDMs override
+    std::vector<RDMs> transition_rdms(const std::vector<std::pair<size_t, size_t>>& root_list,
+                                      std::shared_ptr<ActiveSpaceMethod> method2,
+                                      int max_rdm_level) override;
+
+    /// Set options override
+    void set_options(std::shared_ptr<ForteOptions> options) override;
+
+    /// Quiet mode (no printing, for use with CASSCF)
+    void set_quite_mode(bool quiet) { quiet_ = quiet; }
+
+    /// Set projected roots
+    void project_roots(std::vector<std::vector<std::pair<size_t, double>>>& projected) {
+        projected_roots_ = projected;
+    }
+
+    /// Set initial guess
+    void set_initial_guess(std::vector<std::vector<std::pair<size_t, double>>>& guess) {
+        initial_guess_ = guess;
+    }
+
+    /// Dump wave function to disk
+    void dump_wave_function(const std::string& filename) override;
+
+    /// Read wave function from disk as initial guess
+    bool read_wave_function(const std::string& filename) override;
+
+  private:
+    /// SCFInfo object
+    std::shared_ptr<SCFInfo> scf_info_;
+
+    /// ForteOptions
+    std::shared_ptr<ForteOptions> options_;
+
+    /// Start up function
+    void startup();
+
+    /// Quiet printing
+    bool quiet_ = false;
+
+    /// Number of active orbitals
+    int nactv_;
+    /// Number of active orbitals per irrep
+    psi::Dimension actv_dim_;
+
+    /// Active space type
+    std::string actv_space_type_;
+    /// CI_Reference object
+    std::shared_ptr<CI_Reference> ci_ref_;
+    /// The determinant space
+    DeterminantHashVec p_space_;
+    /// Build determinant space
+    void build_determinant_space();
+
+    /// State label
+    std::string state_label_;
+    /// Multiplicity
```

Comment:
Use the state information object instead of these.

Reply:
These are actually read from `state_` ([here](https://github.com/evangelistalab/forte/pull/227/files/11248643ecc15886d5e279b140bd25d65e44708c#diff-8c55bdbc7eee5cf3882609c3a7791672ea21bf8744d6a96d0d141fb8c575edb0R38)) from `ActiveSpaceMethod`.